internal/service.binding/v1alpha2/cluster_workload_resource_mapping.go

```diff
@@ -18,24 +18,49 @@ package v1alpha2
 
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-// ClusterWorkloadResourceMappingVersion defines the mapping for a specific version of a workload resource.
-type ClusterWorkloadResourceMappingVersion struct {
+// ClusterWorkloadResourceMappingTemplate defines the mapping for a specific version of an workload resource to a
+// logical PodTemplateSpec-like structure.
+type ClusterWorkloadResourceMappingTemplate struct {
 	// Version is the version of the workload resource that this mapping is for.
 	Version string `json:"version"`
-	// Containers is the collection of JSONPaths that container configuration may be written to.
-	Containers []string `json:"containers,omitempty"`
-	// Envs is the collection of JSONPaths that env configuration may be written to.
-	Envs []string `json:"envs,omitempty"`
-	// VolumeMounts is the collection of JSONPaths that volume mount configuration may be written to.
-	VolumeMounts []string `json:"volumeMounts,omitempty"`
-	// Volumes is the JSONPath that volume configuration must be written to.
+	// Annotations is a JSON Pointer that references the annotations map within the workload resource. These
+	// annotations must end up in the resulting Pod, and are generally not the workload resource's annotations.
+	// Defaults to `/spec/template/metadata/annotations`.
+	Annotations string `json:"annotations"`
```

Comment:
this is not present in resource type schema above. Also, can we have an example describing the sage?

Reply:
It has no direct semantic meaning, other than we have found using annotations on the workload to be a useful way to indicate that a particular env var or volume(mount) was created by a service binding. This makes it much easier to remove a binding, or update an existing binding since we can distinguish a user's original intent from an injected value.

Reply:
I removed this, and then immediately remembered why it actually matters (brains are weird like that). When a ServiceBinding defines a custom `type` or `provider` we either need to create a new secret with these values layered in, or we can use a projected volume composed from the secret and fields set with the downward api. Annotations are the best place to write a value that can be consumed by the downward api.

Strictly speaking, this behavior isn't mandated by the workload projection section of the spec, but it's better to not need to munge a secret if we can avoid it.